src/Sylius/Component/Addressing/Model/Country.php

```diff
@@ -146,6 +146,22 @@ public function hasProvince(ProvinceInterface $province)
     /**
      * {@inheritdoc}
      */
+    public function isEqualTo($data)
+    {
+        if (is_object($data)) {
+            if (!$data instanceof CountryInterface) {
+                return false;
+            }
+
+            return $data->getIsoName() === $this->isoName || $data->getId() === $this->id;
+        }
+
+        return $data === $this->id;
```

Comment:
Sure, you don't compare `int` to `object`, but `1` is still equal to `Country`. IMHO that should return false as it's not really equal.